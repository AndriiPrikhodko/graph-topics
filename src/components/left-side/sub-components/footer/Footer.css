.footer {
    margin-top: auto;
    bottom: 0;
    width: 100%;
    background-color: inherit; /* Adjust this to match your page color schema */
    padding: 10px;
    margin-left: -10px;
    display: flex;
    justify-content: space-evenly;
  }
  
  .legal-notice {
    font-family: math;
    width: 90%;
  }
  
  .legal-notice h2 {
    color: ivory;
  }
  
  .legal-notice {
    color: silver;
  }

  .footer a {
    color: darkblue;
    text-decoration: none;
    transition: color 0.3s ease;
  }
  
  .footer a:hover {
    color: darkblue;
    text-decoration: underline;
  }
