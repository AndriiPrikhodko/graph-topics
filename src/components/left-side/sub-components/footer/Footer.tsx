import './Footer.css'

const Footer = () => {
    const currentYear = new Date().getFullYear()

    return (
        <footer className="footer" data-testid='footer'>
            <div className="legal-notice">
                <h2>Author</h2>
                <p>Â© {currentYear} Andrii Prykhodko. All Rights Reserved. email:
                    <a>prykhodkoaa@gmail.com</a>
                </p>
            </div>
        </footer>
    )
}

export default Footer