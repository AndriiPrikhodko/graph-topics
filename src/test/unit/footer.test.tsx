import {  test } from 'vitest'
import { render } from '@testing-library/react'
import Footer from '../../components/left-side/sub-components/footer/Footer'

test('footer', () => {
    const { getByTestId } = render(<Footer />)
    const footerEl = getByTestId('footer')
    expect(footerEl).toMatchSnapshot()
})

