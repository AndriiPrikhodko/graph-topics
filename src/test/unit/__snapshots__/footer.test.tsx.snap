// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`footer 1`] = `
<footer
  class="footer"
  data-testid="footer"
>
  <div
    class="legal-notice"
  >
    <h2>
      Author
    </h2>
    <p>
      Â© 
      2023
       Andrii Prykhodko. All Rights Reserved. email:
      <a>
        prykhodkoaa@gmail.com
      </a>
    </p>
  </div>
</footer>
`;
